src/components/common/modal/index.tsx

```diff
@@ -1,5 +1,49 @@
-const Modal = () => {
-  return <div></div>;
+import { CSSProperties, ReactNode, useMemo, useState } from 'react';
+import * as style from './style.css';
+import useClickAway from '@/hooks/useClickAway';
+import ModalPortal from './ModalPortal';
+
+export type ModalProps = {
+  children: ReactNode;
+  isOpen: boolean;
+  onClose: () => void;
+  type: 'login' | 'icon';
+  style?: CSSProperties;
+};
```

Comment:
```suggestion
export type ModalProps = {
  children: ReactNode;
  type: 'login' | 'icon';
  isOpen: boolean;
  style?: CSSProperties;
  onClose: () => void;
};
```
이 순서가 정답인지 아닌지는 모르겠지만 위에는 값, 아래는 함수로 정렬하면 좋을 것 같아요!